### PR TITLE
[composite compliance] check output of forward-ad with subclass args against regular tensor

### PR DESCRIPTION
**Pull request by kshitij12345**
Time Before PR
```
= 880 passed, 274 skipped, 38170 deselected, 17 xfailed, 21 warnings in 808.96s (0:13:28) =
```

Time After PR
```
= 875 passed, 274 skipped, 38170 deselected, 22 xfailed, 21 warnings in 880.61s (0:14:40) =
```

### CODE REVIEW

**Comment by kshitij12345** [comment]
Margin for failures for `max_unpool{1,2,3}d` was large (have added expected failure):

```
==================================================================== FAILURES =====================================================================
________________________________ TestCompositeComplianceCPU.test_forward_ad_nn_functional_max_unpool1d_cpu_float32 ________________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_functorch/test/test_ops.py", line 1254, in test_forward_ad
    composite_compliance.check_forward_ad_formula(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/composite_compliance.py", line 568, in check_forward_ad_formula
    torch.testing.assert_close(actual_tangents, expected_tangents, equal_nan=True)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1342, in assert_close
    assert_equal(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 1 / 6 (16.7%)
Greatest absolute difference: 0.17576727271080017 at index (0, 1, 1) (up to 1e-05 allowed)
Greatest relative difference: 0.265431923744181 at index (0, 1, 1) (up to 1.3e-06 allowed)
________________________________ TestCompositeComplianceCPU.test_forward_ad_nn_functional_max_unpool2d_cpu_float32 ________________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_functorch/test/test_ops.py", line 1254, in test_forward_ad
    composite_compliance.check_forward_ad_formula(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/composite_compliance.py", line 568, in check_forward_ad_formula
    torch.testing.assert_close(actual_tangents, expected_tangents, equal_nan=True)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1342, in assert_close
    assert_equal(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 1 / 36 (2.8%)
Greatest absolute difference: 1.2434410899877548 at index (0, 0, 1, 4) (up to 1e-05 allowed)
Greatest relative difference: 0.8591214039403613 at index (0, 0, 1, 4) (up to 1.3e-06 allowed)
________________________________ TestCompositeComplianceCPU.test_forward_ad_nn_functional_max_unpool3d_cpu_float32 ________________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_functorch/test/test_ops.py", line 1254, in test_forward_ad
    composite_compliance.check_forward_ad_formula(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/composite_compliance.py", line 568, in check_forward_ad_formula
    torch.testing.assert_close(actual_tangents, expected_tangents, equal_nan=True)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1342, in assert_close
    assert_equal(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 3 / 180 (1.7%)
Greatest absolute difference: 1.3193111419677734 at index (0, 1, 1, 0, 2) (up to 1e-05 allowed)
Greatest relative difference: 1.5144047745409281 at index (0, 1, 2, 4, 2) (up to 1.3e-06 allowed)
_______________________________ TestCompositeComplianceCUDA.test_forward_ad_nn_functional_max_unpool2d_cuda_float32 _______________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_functorch/test/test_ops.py", line 1254, in test_forward_ad
    composite_compliance.check_forward_ad_formula(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/composite_compliance.py", line 568, in check_forward_ad_formula
    torch.testing.assert_close(actual_tangents, expected_tangents, equal_nan=True)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1342, in assert_close
    assert_equal(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 1 / 72 (1.4%)
Greatest absolute difference: 1.0944017171859741 at index (0, 1, 1, 5) (up to 1e-05 allowed)
Greatest relative difference: 1.7053427126956218 at index (0, 1, 1, 5) (up to 1.3e-06 allowed)
_______________________________ TestCompositeComplianceCUDA.test_forward_ad_nn_functional_max_unpool3d_cuda_float32 _______________________________
Traceback (most recent call last):
  File "/home/kshiteej/Pytorch/pytorch_functorch/test/test_ops.py", line 1254, in test_forward_ad
    composite_compliance.check_forward_ad_formula(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_internal/composite_compliance.py", line 568, in check_forward_ad_formula
    torch.testing.assert_close(actual_tangents, expected_tangents, equal_nan=True)
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1342, in assert_close
    assert_equal(
  File "/home/kshiteej/.conda/envs/pytorch-cuda-dev/lib/python3.10/site-packages/torch/testing/_comparison.py", line 1093, in assert_equal
    raise error_metas[0].to_error(msg)
AssertionError: Tensor-likes are not close!

Mismatched elements: 1 / 180 (0.6%)
Greatest absolute difference: 1.7890723645687103 at index (0, 1, 1, 1, 3) (up to 1e-05 allowed)
Greatest relative difference: 3.7477135730390247 at index (0, 1, 1, 1, 3) (up to 1.3e-06 allowed)
================================================================ warnings summary ================================================================
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -13788,7 +13788,10 @@ def error_inputs_mean(op_info, device, **kwargs):
                    'TestCommon', 'test_variant_consistency_eager', device_type='cuda'),
                DecorateInfo(
                    toleranceOverride({torch.float32: tol(atol=1.3e-04, rtol=1.3e-06), }),
-                   'TestCommon', 'test_noncontiguous_samples', device_type='cuda')],
+                   'TestCommon', 'test_noncontiguous_samples', device_type='cuda'),
+               DecorateInfo(
+                   toleranceOverride({torch.float32: tol(atol=1.3e-05, rtol=1.3e-05), }),
+                   'TestCompositeCompliance', 'test_forward_ad', device_type='cuda')],
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:13794]: Tolerance relaxation for `conv_transpose3d`.

**Comment by kshitij12345** [comment]
Ping @zou3519 :)

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**